src/parser/flow_lexer.ml

```diff
@@ -724,65 +767,129 @@ let token (env: Lex_env.t) lexbuf : result =
       is_tail
     ))
 
+
+  | binbigint, word ->
+    (* Numbers cannot be immediately followed by words *)
+    recover env lexbuf ~f:(fun _ _ -> match%sedlex lexbuf with
+    | binbigint -> 
+      Token (env, T_BIGINT { kind = BINARY; raw = lexeme lexbuf })
+    | _ -> failwith "unreachable"
+    )
+  | binbigint ->
+    Token (env, T_BIGINT { kind = BINARY; raw = lexeme lexbuf })
   | binnumber, (letter | '2'..'9'), Star alphanumeric ->
     (* Numbers cannot be immediately followed by words *)
-    recover env lexbuf ~f:(fun env lexbuf -> match%sedlex lexbuf with
+    recover env lexbuf ~f:(fun _ _ -> match%sedlex lexbuf with
     | binnumber ->
-      Token (env, T_NUMBER { kind = BINARY; raw = lexeme lexbuf })
+      Token (env, T_BIGINT { kind = BINARY; raw = lexeme lexbuf })
```

Comment:
This should be `T_NUMBER`

Reply:
I suppose we must be missing a test that looks for `0b123hello` or something like that. Would you mind adding one?